src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs

```diff
@@ -496,13 +496,12 @@ internal static IEnumerable<TrainTestData> CrossValidationSplit(IHostEnvironment
         internal static void EnsureGroupPreservationColumn(IHostEnvironment env, ref IDataView data, ref string samplingKeyColumn, int? seed = null)
         {
             Contracts.CheckValue(env, nameof(env));
-            var host = env.Register("rand");
             // We need to handle two cases: if samplingKeyColumn is provided, we use hashJoin to
             // build a single hash of it. If it is not, we generate a random number.
             if (samplingKeyColumn == null)
             {
                 samplingKeyColumn = data.Schema.GetTempColumnName("SamplingKeyColumn");
-                data = new GenerateNumberTransform(env, data, samplingKeyColumn, (uint?)(seed ?? host.Rand.Next()));
+                data = new GenerateNumberTransform(env, data, samplingKeyColumn, (uint?)(seed ?? env.Seed));
```

Comment:
@justinormont this is where I need `IHostEnvironment.Seed` (this is not public in `MLContext`). Any suggestion on how to get seed without adding it to the interface?

Reply:
Would making it non-public work? Or does that still count as modifying the interface?

@eerhardt, do you see how to let other (internal) ML․NET components access the seed value without modifying the interface?

Reply:
No, the interface members must be public. It can be accessed from `MLContext` after casting it to `IHostEnvironment`, but not without the cast.

One thought is to add an interface that inherits from `IHostEnvironment` and adds a seed. Then, we pass this interface through the code path to `EnsureGroupPreservationColumn` instead of `IHostEnvironment`. Wondering if there is a better way to do this.

Reply:
You can add a new interface. One option is to make that new interface internal. See the usages of

https://github.com/dotnet/machinelearning/blob/de5d48a959a789553c337a7d096e58ef0a2410bb/src/Microsoft.ML.Core/Data/IHostEnvironment.cs#L71-L83

for an example.

Reply:
@justinormont @eerhardt does the latest commit look good?

Reply:
@eerhardt would be the expert here. I would bow to his opinion, though looks good to me.

From @eerhardt's `ICancelable` example I was envisioning a multi-inheritance of having `IHostEnvironment` inherit from a new `ISeeded`, though this may count as modifying `IHostEnvironment`; hence you created a new interface to inherit from `IHostEnvironment`.

New interface:
https://github.com/dotnet/machinelearning/blob/9fd29aca95da8f4359c1ef551ce4f20050327da6/src/Microsoft.ML.Core/Data/IHostEnvironment.cs#L85-L92

Reply:
> I was envisioning a multi-inheritance of having IHostEnvironment inherit from a new ISeeded, though this may count as modifying IHostEnvironment;

Correct - that would count as modifying IHostEnvironment, which is a breaking change.